conformance/tests/httproute-https-listener.go

```diff
@@ -60,10 +60,14 @@ var HTTPRouteHTTPSListener = suite.ConformanceTest{
 			host       string
 			statusCode int
 			backend    string
+			serverName string
 		}{
-			{host: "example.org", statusCode: 200, backend: "infra-backend-v1"},
-			{host: "unknown-example.org", statusCode: 404},
-			{host: "second-example.org", statusCode: 200, backend: "infra-backend-v2"},
+			{host: "example.org", serverName: "example.org", statusCode: 200, backend: "infra-backend-v1"},
+			{host: "example.org", serverName: "second-example.org", statusCode: 421},
+			{host: "second-example.org", serverName: "example.org", statusCode: 421},
```

Comment:
I think the relevant part of the spec is here: https://github.com/kubernetes-sigs/gateway-api/blob/f95fe2b031501392d91d397b458007277d5882b2/apis/v1/gateway_types.go#L394

We're actually using [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119) interpretation of keywords here, so SHOULD is a recommendation, not a requirement. With that said, I think we could justify an extended test (separate feature name) to cover this.

cc @youngnick @rikatz

Reply:
Shouldn't we enforce this behaviour only for reused connections? If Cx is estabilishing a new connection per request then misdirected does not make sense.

@snorwin can you verify that connection is being reused in this test? I was looking into RoundTripper and I can see that [DisableKeepAlives](https://github.com/kubernetes-sigs/gateway-api/blob/f95fe2b031501392d91d397b458007277d5882b2/conformance/utils/roundtripper/roundtripper.go#L140) is set.

Reply:
I understand your example. Let’s assume we have 2 listeners on the same port one with `*.example.com` and second one with `foo.example.com`.

What HTTP error code would you expect for a request on a new connection with SNI `bar.example.com` and a Host header  `foo.example.com`?

Reply:
I think we should update the spec to clarify this as well but LGTM. Thanks!